Sample-App/app/src/main/java/com/adobe/marketing/mobile/sampleapp/EdgeIdentityTab.java

```diff
@@ -136,4 +222,85 @@ public void run() {
             }
         });
     }
-}
\ No newline at end of file
+
+    /**
+     * Async method that retrieves the ad ID from the {@link AdvertisingIdClient} (using Google's gms.ads SDK).
+     * Sanitizes ad ID value to the empty string ({@code ""}) if ad tracking is disabled, or an exception is encountered.
+     *
+     * @param callback receives the ad ID {@code String} if a valid value can be retrieved, {@code ""} otherwise.
+     */
+    private void getAdvertisingIdentifier(final AdobeCallback<String> callback) {
+        if (callback == null) {
+            Log.e(LOG_TAG, "Unexpected null callback, provide a callback to retrieve current GAID.");
+            return;
+        }
+        getAdvertisingIdClientInfo(new AdobeCallback<AdvertisingIdClient.Info>() {
+            @Override
+            public void call(AdvertisingIdClient.Info info) {
+                if (info == null) {
+                    callback.call("");
+                    return;
+                }
+                if (info.isLimitAdTrackingEnabled()) {
+                    callback.call("");
+                } else {
+                    callback.call(info.getId());
+                }
+            }
+        });
+    }
+
+    /**
+     * Async method that retrieves the ad ID from the {@link AdvertisingIdClient} (using Google's gms.ads SDK).
+     * Sanitizes ad ID value to the empty string ({@code ""}) if ad tracking is disabled, or an exception is encountered.
+     *
+     * @param callback receives the ad ID {@code String} if a valid value can be retrieved, {@code ""} otherwise.
+     */
+    private void getIsLimitAdTrackingEnabled(final AdobeCallback<Boolean> callback) {
+        if (callback == null) {
+            Log.e(LOG_TAG, "Unexpected null callback, provide a callback to retrieve isLimitAdTrackingEnabled.");
+            return;
+        }
+        getAdvertisingIdClientInfo(new AdobeCallback<AdvertisingIdClient.Info>() {
+            @Override
+            public void call(AdvertisingIdClient.Info info) {
+                if (info == null) {
+                    return;
+                }
+                callback.call(info.isLimitAdTrackingEnabled());
+            }
+        });
+    }
+
+    /**
+     * Async method that retrieves the {@link AdvertisingIdClient.Info} (using Google's gms.ads SDK).
+     * Callers <strong>MUST</strong> verify that the result of the callback is not null before using any of its properties.
+     *
+     * @param callback receives the {@link AdvertisingIdClient.Info} if a valid value can be retrieved, {@code null} otherwise.
+     */
+    private void getAdvertisingIdClientInfo(final AdobeCallback<AdvertisingIdClient.Info> callback) {
+        if (callback == null) {
+            Log.e(LOG_TAG, "Unexpected null callback, provide a callback to retrieve AdvertisingIdClientInfo.");
```

Comment:
Use `Log.d()` as this is a user implementation issue.

Reply:
Thank you! Updated to use log level `d`